### PR TITLE
Few random improvements related to getting a `DiagnosticWithLocation`

### PR DESCRIPTION
**Pull request by elibarzilay**

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -3696,6 +3696,7 @@ namespace ts {
             getDiagnostics,
         };
 
+        function lookup(diagnostic: DiagnosticWithLocation): DiagnosticWithLocation | undefined;
```
Comment (RyanCavanaugh) [inline src/compiler/utilities.ts:3699]: It doesn't seem like you intended to make a non-overloaded function here?

Comment (elibarzilay) [reply src/compiler/utilities.ts]: Yeah, that was a mistake, finally resolved it.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -1071,13 +1069,15 @@ namespace ts {
...
             const diagnostic = error(location, message, arg0, arg1, arg2, arg3);
             diagnostic.skippedOn = key;
             return diagnostic;
         }
 
-        function error(location: Node | undefined, message: DiagnosticMessage, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): Diagnostic {
+        function error(location: Node, message: DiagnosticMessage, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): DiagnosticWithLocation;
+        function error(location: Node | undefined, message: DiagnosticMessage, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number): Diagnostic;
+        function error(location: Node | undefined, message: DiagnosticMessage, arg0?: string | number, arg1?: string | number, arg2?: string | number, arg3?: string | number) {
             const diagnostic = location
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts:1081]: Possibly subject to the same treatment?

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by elibarzilay** [comment]
Closed because it conflicted with a recent PR, and I could now no longer find the code that this was supposed to simplify.

**Pull request closed without merging**